Library/Formula/jenkins.rb

```diff
@@ -58,12 +64,12 @@ def caveats; <<-EOS.undent
   test do
     ENV["JENKINS_HOME"] = testpath
     pid = fork do
-      exec "#{bin}/jenkins"
+      exec "java -Djava.io.tmpdir=#{testpath} -jar #{opt_libexec}/jenkins.war"
```

Comment:
@dunn There is a reasonable default but it doesn't happen to be within the sandbox. Many (most?) users will never change `java.io.tmpdir`. Some users expect to change `java.io.tmpdir` by passing the `-D` option to `java`, but that isn't exposed via the shell script created by `bin.write_jar_script()` in this formula. There are probably other ways to override it. IMO having jenkins installed via homebrew use `HOMEBREW_TEMP` as the temp dir adheres to the principle of least astonishment. But I could see someone make the case for the system default value of `java.io.tmpdir` adhering to the principle too.